bluesky/run_engine.py

```diff
@@ -1444,16 +1446,28 @@ async def _run(self):
                     await self._stop_movable_objects(success=True)
                     # Notify Devices of the pause in case they want to
                     # clean up.
+                    # track of any devices refuse to pause
+                    refuse_to_pause = False
                     for obj in self._objs_seen:
```

Comment:
This loop probably needs some extra protections to make sure that we hit every object even if it fails.  I guess we need `except BaseException as e: # stash excption ` + finally re-raise anything that was caught and stashed?